FixDevV3Blobs/FixDevV3Blobs/Program.cs

```diff
@@ -0,0 +1,192 @@
+            var stopwatch = await ProcessBlobList(blobList, search, replace);
+
+            Log(
+                "Processed {0} entries in {1}, processing speed: {2} entries per hour, {3} per 1M. Used {4} tasks.",
+                blobList.Count,
+                stopwatch.Elapsed,
+                blobList.Count / stopwatch.Elapsed.TotalHours,
+                TimeSpan.FromTicks(stopwatch.Elapsed.Ticks * (1000000 / blobList.Count)),
+                MaxTasks);
+
+            return 0;
+        }
+
+        private static void LoadProcessedList()
+        {
+            if (File.Exists(ProcessedListFilename))
+            {
+                var lines = File.ReadAllLines(ProcessedListFilename);
+                ProcessedBlobs.UnionWith(lines);
+            }
+        }
+
+        private static async Task<List<CloudBlockBlob>> LoadBlobList(CloudBlobContainer container)
+        {
+            var blobList = new List<CloudBlockBlob>();
+
+            BlobContinuationToken continuationToken = null;
+            var stopwatch = Stopwatch.StartNew();
+            var n = 5;
+            do
+            {
+                var segment = await container.ListBlobsSegmentedAsync(
+                    prefix: null,
+                    useFlatBlobListing: true,
+                    blobListingDetails: BlobListingDetails.None,
+                    maxResults: 5000,
+                    currentToken: continuationToken,
+                    options: null,
+                    operationContext: null);
+
+                var range = segment.Results.OfType<CloudBlockBlob>().ToList();
+                if (range.Count != segment.Results.Count())
+                {
+                    throw new Exception("Unexpected result item type: " + segment.Results.First(r => r.GetType() != typeof(CloudBlockBlob)).GetType());
+                }
+                blobList.AddRange(range);
+                continuationToken = segment.ContinuationToken;
+                Log("Discovered {0} blobs in {1}", blobList.Count, stopwatch.Elapsed);
+            } while (continuationToken != null);
+            stopwatch.Stop();
+            return blobList;
+        }
+
+        private static async Task<Stopwatch> ProcessBlobList(List<CloudBlockBlob> blobList, string search, string replace)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var tasks = Enumerable.Range(1, MaxTasks).Select(_ => Task.CompletedTask).ToArray();
+            var count = 0;
+            var skipped = 0;
+            foreach (var blob in blobList)
+            {
+                ++count;
+                //await ProcessBlob(blob);
+                if (Processed(blob))
+                {
+                    ++skipped;
+                    continue;
+                }
+                await StartProcessing(tasks, blob, search, replace);
+                if ((count % 1000) == 0)
+                {
+                    Log(
+                        "{0}/{1} (skipped: {4}) {2}, ETA: {3}",
```

Comment:
nit: order params and placeholders, i.e. make `{4}` that last placeholder.
it's easier to read.

alternatively use string interpolation to avoid the mapping completely